src/main/java/org/alliancegenome/curation_api/jobs/executors/AgmDiseaseAnnotationExecutor.java

```diff
@@ -75,11 +77,26 @@ public void runLoad(BulkLoadFile bulkLoadFile) {
 	}
 
 	// Gets called from the API directly
-	public void runLoad(BulkLoadFileHistory history, String speciesName, List<AGMDiseaseAnnotationDTO> annotations, List<Long> curiesAdded) {
+	public APIResponse runLoad(String dataProvider, List<AGMDiseaseAnnotationDTO> annotations) {
+
+		List<Long> annotationIdsLoaded = new ArrayList<>();
+		List<Long> annotationIdsBefore = new ArrayList<>();
+		annotationIdsBefore.addAll(agmDiseaseAnnotationDAO.findAllAnnotationIdsByDataProvider(dataProvider));
+		annotationIdsBefore.removeIf(Objects::isNull);
+		
+		BulkLoadFileHistory history = new BulkLoadFileHistory(annotations.size());
+		runLoad(history, dataProvider, annotations, annotationIdsLoaded);
+		runCleanup(diseaseAnnotationService, history, dataProvider, annotationIdsBefore, annotationIdsLoaded);
```

Comment:
@markquintontulloch the only draw back to this is if someone calls the API with a small set of Annotations then clean runs and deletes everything else. This was why I didn't have this runLoad do a cleanup.

Reply:
I kinda made the assumption that the API was only for bulk updates and not "loading"

Reply:
yeah, I was in two minds about this - maybe one for discussion with the group, but this is how it was previously

Reply:
And again these API were only suppose to serve the Integration tests, but I think that @cmpich ZFIN might be using them to do updates. And before they did not run the cleanup.

Reply:
`runLoad(history, speciesNames, alleles, null, null);` when the list is null no cleanup happens.

Reply:
I thought they did do cleanups (before you refactored the cleanup stuff) - that's certainly how they were working for the integration tests before those got refactored

Reply:
Ya... I guess when the cleanup code went in it did it on both.

Reply:
@oblodgett - cleanup now removed for direct API calls